src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -222,1149 +215,1315 @@ static constexpr const int32_t window_editor_object_selection_animation_divisor[
+                gDropdownItemsFormat[DDIX_FILTER_TT] = STR_TOGGLE_OPTION;
+                gDropdownItemsFormat[DDIX_FILTER_OO] = STR_TOGGLE_OPTION;
+                gDropdownItemsFormat[DDIX_FILTER_CUSTOM] = STR_TOGGLE_OPTION;
+
+                gDropdownItemsArgs[DDIX_FILTER_RCT1] = STR_SCENARIO_CATEGORY_RCT1;
+                gDropdownItemsArgs[DDIX_FILTER_AA] = STR_SCENARIO_CATEGORY_RCT1_AA;
+                gDropdownItemsArgs[DDIX_FILTER_LL] = STR_SCENARIO_CATEGORY_RCT1_LL;
+                gDropdownItemsArgs[DDIX_FILTER_RCT2] = STR_ROLLERCOASTER_TYCOON_2_DROPDOWN;
+                gDropdownItemsArgs[DDIX_FILTER_WW] = STR_OBJECT_FILTER_WW;
+                gDropdownItemsArgs[DDIX_FILTER_TT] = STR_OBJECT_FILTER_TT;
+                gDropdownItemsArgs[DDIX_FILTER_OO] = STR_OBJECT_FILTER_OPENRCT2_OFFICIAL;
+                gDropdownItemsArgs[DDIX_FILTER_CUSTOM] = STR_OBJECT_FILTER_CUSTOM;
+
+                // Track manager cannot select multiple, so only show selection filters if not in track manager
+                if (!(gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER))
+                {
+                    numSelectionItems = 3;
+                    gDropdownItemsFormat[DDIX_FILTER_SEPARATOR] = 0;
+                    gDropdownItemsFormat[DDIX_FILTER_SELECTED] = STR_TOGGLE_OPTION;
+                    gDropdownItemsFormat[DDIX_FILTER_NONSELECTED] = STR_TOGGLE_OPTION;
+                    gDropdownItemsArgs[DDIX_FILTER_SEPARATOR] = STR_NONE;
+                    gDropdownItemsArgs[DDIX_FILTER_SELECTED] = STR_SELECTED_ONLY;
+                    gDropdownItemsArgs[DDIX_FILTER_NONSELECTED] = STR_NON_SELECTED_ONLY;
+                }
 
-    return window;
-}
+                WindowDropdownShowText(
+                    { windowPos.x + widgets[widgetIndex].left, windowPos.y + widgets[widgetIndex].top },
+                    widgets[widgetIndex].height() + 1, colours[widgets[widgetIndex].colour], Dropdown::Flag::StayOpen,
+                    _numSourceGameItems + numSelectionItems);
 
-/**
- *
- *  rct2: 0x006AB199
- */
-static void WindowEditorObjectSelectionClose(rct_window* w)
-{
-    unload_unselected_objects();
-    EditorLoadSelectedObjects();
-    editor_object_flags_free();
+                for (int32_t i = 0; i < _numSourceGameItems; i++)
+                {
+                    if (_filter_flags & (1 << i))
+                    {
+                        Dropdown::SetChecked(i, true);
+                    }
+                }
 
-    if (_loadedObject != nullptr)
-        _loadedObject->Unload();
+                if (!(gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER))
+                {
+                    Dropdown::SetChecked(DDIX_FILTER_SELECTED, _FILTER_SELECTED != 0);
+                    Dropdown::SetChecked(DDIX_FILTER_NONSELECTED, _FILTER_NONSELECTED != 0);
+                }
+                break;
+        }
+    }
 
-    if (gScreenFlags & SCREEN_FLAGS_EDITOR)
+    void OnDropdown(rct_widgetindex widgetIndex, int32_t dropdownIndex)
```

Comment:
ci's have noticed that you haven't marked this as `override`

Reply:
oops. done https://github.com/OpenRCT2/OpenRCT2/pull/16027/commits/ed8c45876a851432b94ee8079e5b43d056dc9d48